src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -473,31 +468,20 @@ public void shutdown() {
   private synchronized KafkaConsumerState getConsumerInstance(
       String group,
       String instance,
-      boolean remove
+      boolean toRemove
   ) {
     ConsumerInstanceId id = new ConsumerInstanceId(group, instance);
-    final KafkaConsumerState state = remove ? consumers.remove(id) : consumers.get(id);
+    final KafkaConsumerState state = toRemove ? consumers.remove(id) : consumers.get(id);
     if (state == null) {
       throw Errors.consumerInstanceNotFoundException();
     }
-    // Clear from the timeout queue immediately so it isn't removed during the read operation,
-    // but don't update the timeout until we finish the read since that can significantly affect
-    // the timeout.
-    consumersByExpiration.remove(state);
```

Comment:
The effect of this previously was to prevent expiration until the request had been handled. Should we have some additional logic to accomplish the same thing? Maybe it's good enough if we update the expiration time both when a request is received and when the response is sent?

Reply:
Oh I see... yeah we could update the expiry here as well

Reply:
I think the two cases are offset commits and reads. Both have some logic to update the expiration time after completion, so that's probably good enough.